client/components/withAuth.jsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import { verify } from "../utils/apiWrapper";
+import NavBar from "./NavBar";
+import Router from "next/router";
+import { setCookie } from "./../utils/cookie";
+
+const SUCCESS = 200;
+
+const withAuth = WrappedComponent => {
+  class HOC extends React.Component {
```

Comment:
Could you convert this to functional? We have lots of examples now since the rest of our codebase has been converted!